### PR TITLE
Prevent returning password in reads of connection config info

### PR DESCRIPTION
**Pull request by chrishoffman**

### CODE REVIEW

File: builtin/logical/database/path_config_connection.go [hunk 1]
```diff
@@ -272,7 +271,16 @@ func (b *databaseBackend) connectionWriteHandler() framework.OperationFunc {
 		}
 
 		resp := &logical.Response{}
-		resp.AddWarning("Read access to this endpoint should be controlled via ACLs as it will return the connection details as is, including passwords, if any.")
+
+		// The is a simple test to to check for passwords in the connection_url paramater. If one exists,
+		// warn the user to use templated url string
+		if connURLRaw, ok := config.ConnectionDetails["connection_url"]; ok {
+			if connURL, err := url.Parse(connURLRaw.(string)); err == nil {
```
Comment (jefferai) [inline builtin/logical/database/path_config_connection.go]: On error, shouldn't we check the URL string anyways?

File: builtin/logical/database/path_config_connection.go [hunk 2]
```diff
@@ -272,7 +271,16 @@ func (b *databaseBackend) connectionWriteHandler() framework.OperationFunc {
 		}
 
 		resp := &logical.Response{}
-		resp.AddWarning("Read access to this endpoint should be controlled via ACLs as it will return the connection details as is, including passwords, if any.")
+
+		// The is a simple test to to check for passwords in the connection_url paramater. If one exists,
```
Comment (calvn) [inline builtin/logical/database/path_config_connection.go]: s/The is/This is/

File: builtin/logical/database/path_config_connection.go [hunk 3]
```diff
@@ -272,7 +271,16 @@ func (b *databaseBackend) connectionWriteHandler() framework.OperationFunc {
...
+		// warn the user to use templated url string
+		if connURLRaw, ok := config.ConnectionDetails["connection_url"]; ok {
```
Comment (calvn) [inline builtin/logical/database/path_config_connection.go]: Could we add this check to `connectionReadHandler()` and strip out the `connection_url` if a password is found?

Comment (chrishoffman) [reply builtin/logical/database/path_config_connection.go]: Addressed in https://github.com/hashicorp/vault/pull/4300/commits/0a329a77a5257ab09db73a106821f1893cbbdf92.

File: builtin/logical/database/path_config_connection.go [hunk 4]
```diff
@@ -169,8 +171,18 @@ func (b *databaseBackend) connectionReadHandler() framework.OperationFunc {
...
+			connURL := connURLRaw.(string)
+			if conn, err := url.Parse(connURL); err == nil {
+				if password, ok := conn.User.Password(); ok {
+					config.ConnectionDetails["connection_url"] = strings.Replace(connURL, password, "*****", -1)
+				}
+			}
+		}
+
+		if _, ok := config.ConnectionDetails["password"]; ok {
```
Comment (kalafut) [inline builtin/logical/database/path_config_connection.go]: This can be simply `delete(config.ConnectionDetails, "password")`

**Pull request merged**